clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1589,7 +1607,8 @@ public void handleResponse(AbstractResponse response) {
                 fatalError(error.exception());
             } else if (error == Errors.INVALID_TXN_STATE) {
                 fatalError(error.exception());
-            } else if (error == Errors.UNKNOWN_PRODUCER_ID || error == Errors.INVALID_PRODUCER_ID_MAPPING) {
+            } else if (error == Errors.UNKNOWN_PRODUCER_ID || error == Errors.INVALID_PRODUCER_ID_MAPPING
+                    || error == Errors.TRANSACTION_TIMED_OUT) {
```

Comment:
The documentation suggests that after catching `TransactionTimeoutException`, the user can just begin a new transaction. Unless I'm missing something, however, it seems like we still require an explicit call to `abortTransaction`. That is actually what I prefer, but we should clarify the expectation in the documentation. It is super important for the integration test to cover the full expected flow.

Reply:
Thanks for your advice. I will update doc and the kip later